anonlink/similarities/_dice_x86.py

```diff
@@ -56,32 +66,34 @@ def dice_coefficient_accelerated(
     if k is None or k > length_f1:
         k = length_f1
 
-    filter_bits = len(filters0[0])
-    if (any(len(f) != filter_bits for f in filters0)
-            or any(len(f) != filter_bits for f in filters1)):
+    if not _all_equal(map(len, chain(filters0, filters1))):
         raise ValueError('inconsistent filter length')
+    filter_bits = len(filters0[0])
     if filter_bits % 64:
         msg = (f'only filters whose length is a multiple of 64 are currently '
                f'supported (got filter with length ({filter_bits})')
         raise NotImplementedError(msg)
     filter_bytes = filter_bits // 8
 
-    # An array of the *one* filter
-    clist0 = (ffi.new(f"char[{filter_bytes}]", f.tobytes()) for f in filters0)
+    # Array of the one filter from filters0 at a time.
+    carr0 = ffi.new('char[]', filter_bytes)
+
+    # Array of all filters from filters1.
     carr1 = ffi.new(f"char[{filter_bytes * length_f1}]",
                     b''.join(f.tobytes() for f in filters1))
+
     c_popcounts = ffi.new(f"uint32_t[{length_f1}]",
-                          [f.count() for f in filters1])
+                          tuple(f.count() for f in filters1))
 
     # easier to do all buffer allocations in Python and pass them to C,
     # even for output-only arguments
     c_scores = ffi.new("double[]", k)
     c_indices = ffi.new("int[]", k)
 
-    for i, f0 in enumerate(clist0):
-        assert len(f0) == filter_bytes
+    for i, f0 in enumerate(filters0):
+        carr0[0:filter_bytes] = f0.tobytes()
```

Comment:
You raise a very good point. I haven't run the benchmarks on this but I will now. (This change was only because I thought it was clearer to do things this way.)

Reply:
I've run the benchmarks now. The changes actually result in a small improvement in speed. (E.g., from 101.3 Mhash/s to 105.7 Mhash/s on a big problem).